unumpy/_multimethods.py

```diff
@@ -1421,3 +1564,153 @@ def _vander_default(x, N=None, increasing=False):
 @all_of_type(ndarray)
 def vander(x, N=None, increasing=False):
     return (x,)
+
+
+@create_numpy(_first2argreplacer)
+@all_of_type(ndarray)
+def split(ary, indices_or_sections, axis=0):
+    return (ary, indices_or_sections)
+
+
+@create_numpy(_first2argreplacer)
+@all_of_type(ndarray)
+def array_split(ary, indices_or_sections, axis=0):
+    return (ary, indices_or_sections)
+
+
+@create_numpy(
+    _first2argreplacer,
+    default=lambda ary, indices_or_sections: split(ary, indices_or_sections, axis=2),
+)
+@all_of_type(ndarray)
+def dsplit(ary, indices_or_sections):
+    return (ary, indices_or_sections)
+
+
+@create_numpy(
+    _first2argreplacer,
+    default=lambda ary, indices_or_sections: split(ary, indices_or_sections, axis=1),
+)
+@all_of_type(ndarray)
+def hsplit(ary, indices_or_sections):
+    return (ary, indices_or_sections)
+
+
+@create_numpy(
+    _first2argreplacer,
+    default=lambda ary, indices_or_sections: split(ary, indices_or_sections, axis=0),
+)
+@all_of_type(ndarray)
+def vsplit(ary, indices_or_sections):
+    return (ary, indices_or_sections)
+
+
+def _tile_default(A, reps):
+    if isinstance(reps, numbers.Integral):
+        reps = (reps,)
+
+    prep = len(reps) - ndim(A)
+    if prep > 0:
+        for _ in range(prep):
+            A = expand_dims(A, 0)
+    elif prep < 0:
+        reps = tuple(itertools.repeat(1, abs(prep))) + reps
+
+    for axis, dim in enumerate(reps):
+        dim = operator.index(dim)
+        if dim == 0:
+            return asarray([], dtype=A.dtype)
+
+        tile = A
+        for _ in range(dim - 1):
+            A = concatenate((A, tile), axis=axis)
+
+    return A
+
+
+@create_numpy(_first2argreplacer, default=_tile_default)
+@all_of_type(ndarray)
+def tile(A, reps):
+    return (A, reps)
+
+
+@create_numpy(_first2argreplacer)
+@all_of_type(ndarray)
+def repeat(a, repeats, axis=None):
+    return (a, repeats)
+
+
+@create_numpy(_first2argreplacer)
+@all_of_type(ndarray)
+def delete(arr, obj, axis=None):
+    return (arr, obj)
+
+
+def _insert_argreplacer(args, kwargs, dispatchables):
+    def replacer(arr, obj, values, axis=None):
+        return dispatchables, {"axis": None}
+
+    return replacer(*args, **kwargs)
+
+
+@create_numpy(_insert_argreplacer)
+@all_of_type(ndarray)
+def insert(arr, obj, values, axis=None):
+    return (arr, obj, values)
+
+
+@create_numpy(_first2argreplacer)
+@all_of_type(ndarray)
+def append(arr, values, axis=None):
+    return (arr, values)
+
+
+@create_numpy(_self_argreplacer)
+@all_of_type(ndarray)
+def resize(a, new_shape):
+    return (a,)
+
+
+def _trim_zeros(filt, trim="fb"):
+    if "f" in trim and filt[0] == 0:
+        return _trim_zeros(filt[1:], trim)
+    if "b" in trim and filt[-1] == 0:
+        return _trim_zeros(filt[:-1], trim)
```

Comment:
This is a really inefficient recursive implementation... I'd just remove the default.

Reply:
I think this implementation is more efficient:

```python
def _trim_zeros_default(filt, trim="fb"):
    nonzero_idxs = nonzero(filt)[0]
    remove_idxs = []
    if "f" in trim and nonzero_idxs[0] != 0:
        remove_idxs += list(range(nonzero_idxs[0]))
    if "b" in trim and nonzero_idxs[-1] != len(filt) - 1:
        remove_idxs += list(range(nonzero_idxs[-1] + 1, len(filt)))

    return delete(filt, remove_idxs)
```

Reply:
Still some improvement to be made. :smile: 

I'd replace `range` with `arange` and `+=` with `concatenate` and you have a working implementation!

Reply:
Replace `concatenate` with `append` also?

Reply:
Isn't this just a slice?
```python
def _trim_zeros_default(filt, trim="fb"):
    nonzero_idxs = nonzero(filt)[0]
    start, end = None, None
    if "f" in trim:
        start = nonzero_idxs[0]
    if "b" in trim:
        end = nonzero_idxs[-1] + 1
    return filt[start:end]
```

Reply:
Your solution looks a lot simpler actually. 😅 I think I would only add `asarray` to the return statement so that it returns a new array instead of a view.

Reply:
`np.trim_zeros` does return a view.
```python
In [9]: a = np.arange(10)

In [10]: np.trim_zeros(a)[4] = 0

In [11]: a
Out[11]: array([0, 1, 2, 3, 4, 0, 6, 7, 8, 9])

In [12]: a[5]
Out[12]: 0

```

Reply:
You are right. The data type should also be preserved so `asarray` shouldn't be used.